lua/autorun/ragdollremover.lua

```diff
@@ -11,92 +11,132 @@ local RemoveSpeed = 4000        -- Velocity ragdoll is removed at
-else
-        local IsTTT = false
-end
+local IsTTT = false
+
+hook.Add("InitPostEntity","CheckForTTT", function() -- This is something I'm not quite sure about, but we may run into an issue that when the file is loaded the TTT gamemode hasn't been loaded. This insures that we check after everything has been loaded.
+	IsTTT = GAMEMODE_NAME == "terrortown" and true or false
+end)
  
-local function SetSubPhysMotionEnabled( ent, enable )
-        if not IsValid( ent ) then return end
-	
-        ent:SetVelocity( vector_origin )
+local entMeta = FindMetaTable("Entity") -- Why? Both ways work but I think the looks cleaner :P
+
+function entMeta:SetSubPhysMotion( enable )
+	if not IsValid( self ) then return end
 	
-        if ( not enable ) then
-                ent:SetColor( Color( 255, 0, 255, 255 ) )
-                if IsValid( ent:GetOwner() ) then
-                        if ( IsTTT ) then
-                                ent:GetOwner():GetWeapon( "weapon_zm_carry" ):Reset( false )
-                        end
-                end
-        else
-                ent:SetColor( Color( 255, 255, 255, 255 ) )
-        end
+	if ( not enable ) then
+		self:SetColor( Color( 255, 0, 255, 255 ) )
+		if IsValid( self:GetOwner() ) and IsTTT then
+			self:GetOwner():GetWeapon( "weapon_zm_carry" ):Reset( false )
+		end
+	else
+		self:SetColor( Color( 255, 255, 255, 255 ) )
+	end
 	
-        for i = 0, ent:GetPhysicsObjectCount() - 1 do
-                local subphys = ent:GetPhysicsObjectNum( i )
-                if IsValid( subphys ) then
-                        subphys:EnableMotion( enable )
-                        if ( not enable ) then
-                                subphys:SetVelocity( vector_origin )
-                                subphys:SetMass( subphys:GetMass() * 20 )
-                        else
-                                subphys:SetMass( subphys:GetMass() / 20 )
-                                subphys:Wake()
-                        end
-                end
+	for i = 0, self:GetPhysicsObjectCount() - 1 do
+		local subphys = self:GetPhysicsObjectNum( i )
+		if IsValid( subphys ) then
+			subphys:EnableMotion( enable )
+			if ( not enable ) then
+				subphys:SetVelocity( vector_origin )
+				subphys:SetMass( subphys:GetMass() * 20 )
+			else
+				subphys:SetMass( subphys:GetMass() / 20 )
+				subphys:Wake()
+			end
+		end
 	end
-
-        ent:SetVelocity( vector_origin )
+	self:SetVelocity( vector_origin )
 end
  
 local function KillVelocity( ent )
-	SetSubPhysMotionEnabled( ent, false )
+	if not IsValid(ent) then return end
+	
+	ent:SetSubPhysMotion(false)
 	
 	if ( EchoFreeze ) then 
 		PrintMessage( HUD_PRINTTALK, "[GS_CRASH] Body frozen for " .. FreezeTime .. " seconds!" ) 
 	end
 	
-	timer.Simple( FreezeTime, function() 
-		SetSubPhysMotionEnabled( ent, true )
-		if ( EchoFreeze ) then 
-			PrintMessage( HUD_PRINTTALK, "[GS_CRASH] Body unfrozen!" )
+	timer.Simple(FreezeTime, function()
+		if IsValid(ent) then -- Just in case
```

Comment:
There shouldn't be a case were its not valid. Again its my personal preference, as I tend to error on the side of caution. Because in all reality it should never come up as not valid. But just in case the ragdoll for some reason gets removed, it will insure that the code never errors.